Zend/zend_language_scanner.l

```diff
@@ -1563,6 +1658,15 @@ NEWLINE ("\r"|"\n"|"\r\n")
 
 
 <ST_IN_SCRIPTING>"}" {
+	if(in_class) {
+		if(--in_class == 1) {
+			in_class = 0;	
+		}
+		if(in_use_list) {
+			in_use_list = 0;
+			in_trait_conflict = 0;
+		}
+	}
```

Comment:
What about code like `class A { function foo() { "${$a}"; } function list() {} }`, is this handled correclty? The opening `{` for the string interpolation won't be lexed as a plain curly open brace in this case.